src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1660,11 +1662,30 @@ private void CompileAndExecuteForDeltaExecution(List<AssociativeNode> astList)
         private List<Guid> PreviewInternal(GraphSyncData syncData)
         {
             var previewChangeSetComputer = changeSetComputer.Clone();
+
             // Get the list of ASTs that will be affected by syncData
             var previewAstList = previewChangeSetComputer.GetDeltaASTList(syncData);
 
             // Get the list of guid's affected by the astlist
             List<Guid> cbnGuidList = previewChangeSetComputer.EstimateNodesAffectedByASTList(previewAstList);
+
+            var finalDeltaAstList = new List<AssociativeNode>();
+
+            // Newly added nodes will not be in the VM yet.
+            // So we need to add them to the preview list.
+            var addCSComputer = changeSetComputer.Clone();
+
+            var addedDeltaAsts = addCSComputer.GetDeltaAstListAdded(syncData.AddedSubtrees);
```

Comment:
@pinzart90 can't you use the `previewChangeSetComputer` here, do you need to clone it again?

Reply:
Sadly yes. If I use the `previewChangeSetComputer` then the call to `GetDeltaAstListAdded` will throw an exception with ` same key already exists`